runtime/fundamentals/node.md

```diff
@@ -149,14 +154,32 @@ $ deno run -R -E main.cjs
 ```
 
 `-R` and `-E` flags are used to allow permissions to read files and environment
-variables. permissions.
+variables.
 
-**Deno's permission system is still in effect when using CommonJS modules.** It
-is necessary to provide at least `--allow-read` permission as Deno will probe
-the file system for `package.json` files and `node_modules` directory to
-properly resolve CommonJS modules.
+### package.json type option
+
+Deno will treat files as CommonJS if there's a `package.json` file with
+`"type": "commonjs"` option next to the file, or up in the directory tree and
+`--unstable-detect-cjs` flag is passed on the CLI.
+
+```json title="package.json"
+{
+  "type": "commonjs"
+}
+```
+
+```js title="main.js"
+const express = require("express");
+```
+
+Tools like Next.js's bundler and others will generate a `package.json` file like
+that automatically.
+
+If you have an existing project that uses CommonJS modules, you can make it work
+with both Node.js and Deno, by adding `"type": "commonjs"` option to the
+`package.json` file.
 
-***Create require() manually**
+### Create require() manually
```

Comment:
I'd swapped these to bold rather than titles so that they don't show up in the right hand nav because it was getting a little overwhelming on this page.